aptos-move/block-executor/src/scheduler_v2.rs

```diff
@@ -66,3 +422,1400 @@ impl ExecutionQueueManager {
+    ///     to update the status of the just-executed transaction (e.g., to `Executed` or to
+    ///     `PendingScheduling` if it was aborted concurrently). If this call indicates the
+    ///     transaction is now `Executed` (returns true), `txn_idx` is also added to the
+    ///     `propagation_queue`.
+    /// 5.  **Arm Commit Lock**: If the transaction is now `Executed` and it's either transaction 0
+    ///     or the preceding transaction (`txn_idx - 1`) is no longer `NotCommitted`, it arms
+    ///     the `queueing_commits_lock`, implying that `txn_idx` might be ready for commit.
+    /// 6.  **Update `executed_once_max_idx`**: If `incarnation == 0`, calls
+    ///     `[SchedulerV2::try_increase_executed_once_max_idx]`.
+    /// 7.  **Propagate Stalls/Unstalls**: Calls `[SchedulerV2::propagate]` to recursively
+    ///     process stall/unstall signals based on the status changes of transactions in the
+    ///     `propagation_queue`.
+    ///
+    /// Returns `Err(PanicError)` if any underlying status update fails.
+    pub(crate) fn finish_execution<'a>(
+        &'a self,
+        abort_manager: AbortManager<'a>,
+    ) -> Result<(), PanicError> {
+        let (txn_idx, incarnation, invalidated_set) = abort_manager.take();
+
+        if incarnation > 0 {
+            // Record aborted dependencies. Only recording for incarnations > 0 is in line with the
+            // optimistic value validation principle of Block-STMv2. 0-th incarnation might invalidate
+            // due to the first write, but later incarnations could make the same writes - in which case
+            // there is no need to record (and stall, etc) the corresponding dependency.
+            self.aborted_dependencies[txn_idx as usize]
+                .lock()
+                .record_dependencies(invalidated_set.keys().copied());
+        }
+
+        let mut propagation_queue: BTreeSet<usize> = BTreeSet::new();
+        for (txn_idx, maybe_incarnation) in invalidated_set {
+            if let Some(incarnation) = maybe_incarnation {
+                self.txn_statuses.finish_abort(txn_idx, incarnation, true)?;
+                propagation_queue.insert(txn_idx as usize);
+            }
+        }
+
+        if self.txn_statuses.finish_execution(txn_idx, incarnation)? {
+            propagation_queue.insert(txn_idx as usize);
+
+            if txn_idx == 0
+                || self.committed_marker[txn_idx as usize - 1].load(Ordering::Relaxed)
+                    != CommitMarkerFlag::NotCommitted as u8
+            {
+                // If the committed marker is NOT_COMMITTED by the time the last execution of a
+                // transaction finishes, then considering the lowest such index, arming will occur
+                // either because txn_idx = 0 (base case), or after the marker is set, in the
+                // commits_hooks_unlock method (which checks the executed status).
+                self.queueing_commits_lock.arm();
```

Comment:
What does arm() do?

Reply:
arms the lock xD
if it's armed and unlocked it can be locked.
(implementation is very few lines if you want to look as well)

Reply:
I'll add a comment as well somewhere